### PR TITLE
Submission/epsilon flashonly 2026 04 06

### PR DESCRIPTION
**Pull request by teerthsharma**
# EPSILON FlashOnly 11L: Maximum Expressivity Profile

A high-efficiency, maximum expressivity transformer configuration optimized for a 16MB artifact constraint. This profile utilizes a 11-layer architecture with a hybrid optimization strategy (Muon/AdamW) and custom quantization techniques to achieve a reported **1.11 val_bpb**.

## 🚀 Pipeline Overview

```mermaid
flowchart TD
    %% Global Styles
    classDef data fill:#2d3436,stroke:#636e72,color:#dfe6e9,rx:5,ry:5,stroke-width:1px;
    classDef compute fill:#1e3a8a,stroke:#3b82f6,color:#ffffff,rx:10,ry:10,stroke-width:2px;
    classDef loss fill:#b91c1c,stroke:#ef4444,color:#ffffff,shape:diamond,stroke-width:2px;
    classDef postprocess fill:#701a75,stroke:#c084fc,color:#ffffff,rx:10,ry:10,stroke-width:2px;
    classDef artifact fill:#166534,stroke:#4ade80,color:#ffffff,shape:cylinder,stroke-width:2px;
    classDef eval fill:#92400e,stroke:#f59e0b,color:#ffffff,rx:5,ry:5,stroke-width:1px;

    subgraph STAGE_INPUT ["DATA INGEST"]
        direction LR
        A["FineWeb Shards + SP Tokenizer"]
        B["Distributed Token Loader"]
        A --> B
    end
    class A,B data;

    B ==> STAGE_MODEL

    subgraph STAGE_MODEL ["COMPUTE & ARCHITECTURE"]
        direction TB
        C["GPT 11L 512d 8H/4KV"]
        C --> D["FlashAttention Only"]
        C --> E["XSA on Last 11 Layers"]
        C --> F["BigramHash + SmearGate + VE"]
    end
    class C,D,E,F compute;

    D --> G
    E --> G
    F --> G
    G["Cross-Entropy Loss"]:::loss

    G ==> STAGE_POST

    subgraph STAGE_POST ["POST-TRAINING OPTIMIZATION"]
        direction TB
        H["Muon + AdamW Split"]
        I["EMA + SWA"]
        J["GPTQ Int6 + Selective Prune"]
        H --> I
        I --> J
    end
    class H,I,J postprocess;

    J ==> K["LZMA Artifact (15.9 MB)"]:::artifact
    K --> L["Val Eval + Sliding Window"]:::eval

    %% Link Styling
    linkStyle default stroke:#a1a1aa,stroke-width:1px;
    linkStyle 2,7,10 stroke:#60a5fa,stroke-width:4px;

    style STAGE_INPUT fill:#111,stroke:#636e72,color:#fff;
    style STAGE_MODEL fill:#111,stroke:#3b82f6,color:#fff;
    style STAGE_POST fill:#111,stroke:#c084fc,color:#fff;

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — Submission/epsilon flashonly 2026 04 06

**BPB:** (not parsed — see PR title) | **Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache

**What I found in the code** (head SHA `0f3c7231323e`, file `records/track_10min_16mb/2026-03-22_AETHER_SparseSDP_Governor/train_gpt.py`):

Static code review found no TTT adaptation function, no SLOT optimization loop, no n-gram-cache class, and no pre-quant val-token fine-tune. The eval path uses the standard sliding-window stride-64 pattern. The submission is a pure-neural architecture iteration on the standard SP1024/SP4096/SP8192 baseline.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 0.11s, dim=512, layers=12, vocab=1024, code=61637 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks (3-seed validation, under-16MB artifact cap, ≤600s train + ≤600s eval on 8×H100 SXM). No compliance flags from the classification pass — this looks like a clean pure-neural iteration on the standard baseline.

*Auto-classification caveat:* this review was drafted by the AST-based classifier. If there's a non-standard eval mechanism (logit postprocessing, hedge mixing, etc.) that I missed because it's factored into a helper file or a non-standard function name, please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 0.11s, dim=512, layers=12, vocab=1024, code=61637 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*
